magefiles/rulesengine/repos/infra_deployments.go

```diff
@@ -1,17 +1,152 @@
+		AddLabelToLabelFilter(rctx, "integration-service")
+		return nil
+	})}}
+
+var InfraDeploymentsEnterpriseControllerTestFileChangeRule = rulesengine.Rule{Name: "Infra-deployments PR Enterprise Controller component File Change Rule",
+	Description: "Map Enterprise Controller tests files when EC component files are changed in the infra-deployments PR",
+	Condition: rulesengine.ConditionFunc(func(rctx *rulesengine.RuleCtx) bool {
+
+		return len(rctx.DiffFiles.FilterByDirGlob("components/enterprise-contract/**/*")) != 0
+	}),
+	Actions: []rulesengine.Action{rulesengine.ActionFunc(func(rctx *rulesengine.RuleCtx) error {
+		AddLabelToLabelFilter(rctx, "ec")
+		return nil
+	})}}
+
+var InfraDeploymentsJVMTestFileChangeRule = rulesengine.Rule{Name: "Infra-deployments PR Jvm-build-service component File Change Rule",
+	Description: "Map jvm-build-service tests files when Jvm-build-service component files are changed in the infra-deployments PR",
+	Condition: rulesengine.ConditionFunc(func(rctx *rulesengine.RuleCtx) bool {
+
+		return len(rctx.DiffFiles.FilterByDirGlob("components/jvm-build-service/**/*")) != 0
+	}),
+	Actions: []rulesengine.Action{rulesengine.ActionFunc(func(rctx *rulesengine.RuleCtx) error {
+		AddLabelToLabelFilter(rctx, "jvm-build-service")
+		return nil
+	})}}
+
+var InfraDeploymentsImageControllerTestFileChangeRule = rulesengine.Rule{Name: "Infra-deployments PR Image Controller component File Change Rule",
+	Description: "Map image-controller tests files when Image Controller component files are changed in the infra-deployments PR",
+	Condition: rulesengine.ConditionFunc(func(rctx *rulesengine.RuleCtx) bool {
+
+		return len(rctx.DiffFiles.FilterByDirGlob("components/image-controller/**/*")) != 0
+	}),
+	Actions: []rulesengine.Action{rulesengine.ActionFunc(func(rctx *rulesengine.RuleCtx) error {
+		AddLabelToLabelFilter(rctx, "image-controller")
+		return nil
+	})}}
+
+var InfraDeploymentsMultiPlatformTestFileChangeRule = rulesengine.Rule{Name: "Infra-deployments PR Multi Controller component File Change Rule",
+	Description: "Map multi platform tests files when Multi Controller component files are changed in the infra-deployments PR",
+	Condition: rulesengine.ConditionFunc(func(rctx *rulesengine.RuleCtx) bool {
+
+		return len(rctx.DiffFiles.FilterByDirGlob("components/multi-platform-controller/**/*")) != 0
+	}),
+	Actions: []rulesengine.Action{rulesengine.ActionFunc(func(rctx *rulesengine.RuleCtx) error {
+		AddLabelToLabelFilter(rctx, "multi-platform")
+		return nil
+	})}}
+
+var InfraDeploymentsBuildTemplatesTestFileChangeRule = rulesengine.Rule{Name: "Infra-deployments PR Build-templates component File Change Rule",
+	Description: "Map build-templates tests files when Build-templates component files are changed in the infra-deployments PR",
+	Condition: rulesengine.ConditionFunc(func(rctx *rulesengine.RuleCtx) bool {
+
+		return len(rctx.DiffFiles.FilterByDirGlob("components/build-templates/base/**/*")) != 0
```

Comment:
shouldn't it be `"components/build-templates/**/*"`?

...or when I think about it, it would maybe make more sense to trigger `build-templates` tests when the [pipeline configmap is updated](https://github.com/redhat-appstudio/infra-deployments/tree/main/components/build-service/base/build-pipeline-config) instead

@tisutisu wdyt?

Reply:
@psturc  yes you are right. build-definitions changes are promoted through [this ConfigMap](https://github.com/redhat-appstudio/infra-deployments/tree/main/components/build-service/base/build-pipeline-config), it would be good to run `build-templates` labelled tests when `build-pipeline-config.yaml` file is updated.

Reply:
then for **build templates**  Rule the condition should be like this 
```suggestion
		return len(rctx.DiffFiles.FilterByDirGlob("components/build-service/base/**/*")) != 0
```
if any changes was made in [components/build-templates](https://github.com/redhat-appstudio/infra-deployments/tree/main/components/build-templates) trigger any specific suite or the default konflux is fine?

Reply:
So according to what Sushanta mentioned above, I think we should trigger "build-templates" test, when the related config map is updated, i.e.
```
return len(rctx.DiffFiles.FilterByDirGlob("components/build-service/base/build-pipeline-config/*")) != 0
```

And since this would cover the `build-templates` test scenario, I think we don't need another rule for `components/build-templates` and its subdirectories, i.e. the default test suite (`konflux`) should be run.

@tisutisu please correct me if I'm wrong.

Reply:
@psturc I am okay with this change ` we should trigger "build-templates" test, when the related config map is updated and don't need another rule for components/build-templates and its subdirectories, i.e. the default test suite (konflux) should be run`

Reply:
ok, 
based on that , 
 we should  exclude this change from the components/build-service Rule , right ?

Reply:
components/build-service rule will be there..  but needs to be modified.. 
- when this file `components/build-service/base/build-pipeline-config/build-pipeline-config.yaml` changed, run `build-templates` labeled tests
- when changes in other sub-directories inside `components/build-service` except that `build-pipeline-config.yaml`, we can run `build-service` labeled  tests